File: uhrwerk-config/src/main/java/io/qimia/uhrwerk/config/representation/Env.java
package io.qimia.uhrwerk.config.representation;

import java.util.Objects;

public class Env {
  Metastore metastore;

  public Metastore getMetastore() {
    return metastore;
  }

  public void setMetastore(Metastore metastore) {
    this.metastore = metastore;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Env env = (Env) o;
    return Objects.equals(metastore, env.metastore);
  }

  @Override
  public int hashCode() {
    return Objects.hash(metastore);
  }

  @Override
  public String toString() {
    return "Env{" + "metastore=" + metastore + '}';
  }
}

File: uhrwerk-config/src/main/kotlin/io/qimia/uhrwerk/config/representation/Env.kt
package io.qimia.uhrwerk.config.representation

import com.fasterxml.jackson.annotation.JsonProperty
import io.qimia.uhrwerk.config.builders.ConfigException
import java.util.*

data class Env(
    var secrets: Array<Secret>? = null,
    @JsonProperty("meta_store")
    var metastore: Metastore? = null
) {
    fun validate(path: String) {
        var path = path
        path += "env/"
        if (metastore == null) {
            throw ConfigException("Missing field: " + path + "metastore")
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Env) return false

        if (secrets != null) {
            if (other.secrets == null) return false
            if (!secrets.contentEquals(other.secrets)) return false
        } else if (other.secrets != null) return false
        if (metastore != other.metastore) return false

        return true
    }

    override fun hashCode(): Int {
        var result = secrets?.contentHashCode() ?: 0
        result = 31 * result + (metastore?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "Env(secrets=${secrets?.contentToString()}, metastore=$metastore)"
    }
}